data/sea/54-zebra.h

```diff
@@ -74,6 +59,68 @@ ierror_msg_t zebra_snapshot_step
   , zebra_state_t *state
   , zebra_entity_t *entity );
 
+static uint64_t zebra_translate_column
+  ( anemone_mempool_t *mempool
+  , int64_t elem_start
+  , int64_t elem_count
+  , void **dst
+  , const zebra_column_t *src );
+
+
+static uint64_t zebra_translate_table
+    ( anemone_mempool_t *mempool
+    , int64_t elem_start
+    , int64_t elem_count
+    , void **dst
+    , const zebra_table_t *src )
+{
+    const int64_t row_count = src->row_count;
+    const zebra_table_tag_t tag = src->tag;
+    const zebra_table_variant_t data = src->of;
+
+    switch (tag) {
+    case ZEBRA_TABLE_BINARY: {
+            char *bytes = anemone_mempool_alloc(mempool, elem_count + 1);
+            memcpy(bytes, data._binary.bytes + elem_start, elem_count);
+            bytes[elem_count] = '\0';
+            *dst = bytes;
+            return 1;
+        }
+
+        case ZEBRA_TABLE_ARRAY: {
+            int64_t offset = 0;
+            for (int64_t i = 0; i != row_count; ++i) {
+                offset += zebra_translate_column
+                    ( mempool
+                    , elem_start
+                    , elem_count
+                    , dst + offset
+                    , data._array.values + i );
```

Comment:
the row_count here is the number of elements, not the number of columns - the table only has one top-level column, and this loop should be in the ZEBRA_COLUMN_STRUCT or something